lib/reek/smells/smell_repository.rb

```diff
@@ -9,29 +9,37 @@ module Smells
     #
     # @api private
     class SmellRepository
-      attr_reader :detectors
+      attr_reader :detectors, :typed_detectors, :configuration, :source_via, :smell_types
```

Comment:
Are we accessing all these values outside of this class? If not, I don't think we should expand this class' API by adding these readers.